griddb/src/main/java/com/yahoo/ycsb/db/griddb/GridDBClient.java

```diff
@@ -0,0 +1,333 @@
+import com.toshiba.mwcloud.gs.ContainerInfo;
+import com.toshiba.mwcloud.gs.ContainerType;
+import com.toshiba.mwcloud.gs.GSException;
+import com.toshiba.mwcloud.gs.GSType;
+import com.toshiba.mwcloud.gs.GridStore;
+import com.toshiba.mwcloud.gs.GridStoreFactory;
+import com.toshiba.mwcloud.gs.PartitionController;
+import com.toshiba.mwcloud.gs.Row;
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+
+/**
+ * A class representing GridDBClient.
+ *
+ */
+public class GridDBClient extends com.yahoo.ycsb.DB {
+  //(A)multicast method
+  private String notificationAddress = ""; // "239.0.0.1"
+  private String notificationPort = ""; // "31999"
+  //(B)fixed list method
+  private String notificationMember = ""; // "10.0.0.12:10001,10.0.0.13:10001,10.0.0.14:10001"
+
+  private String userName = ""; //
+  private String password = ""; //
+  private String clusterName = ""; // "ycsbCluster"
+
+
+  public static final String VALUE_COLUMN_NAME_PREFIX= "field";
+  public static final int ROW_KEY_COLUMN_POS = 0;
+
+  private String containerPrefix = "";
+
+  public static final int DEFAULT_CACHE_CONTAINER_NUM = 1000;
+  public static final int FIELD_NUM = 10;
+
+  private int numContainer = 0; // Sets PartitionNum
+  public static final GSType SCHEMA_TYPE = GSType.STRING;
+
+  private GridStore store;
+  private ContainerInfo containerInfo = null;
+  private static final Logger LOGGER = Logger.getLogger(GridDBClient.class.getName());
+
+  class RowComparator implements Comparator<Row> {
+    public int compare(Row row1, Row row2) throws NullPointerException {
+      int result = 0;
+      try {
+        Object val1 = row1.getValue(0);
+        Object val2 = row2.getValue(0);
+        result = ((String)val1).compareTo((String)val2);
+      } catch (GSException e) {
+        LOGGER.severe("There is a exception: " + e.getMessage());
+        throw new NullPointerException();
+      }
+      return result;
+    }
+  }
+
+  public void init() throws DBException {
+    LOGGER.info("GridDBClient");
+
+    final Properties props = getProperties();
+    notificationAddress = props.getProperty("notificationAddress");
+    notificationPort = props.getProperty("notificationPort");
+    notificationMember = props.getProperty("notificationMember");
+    clusterName = props.getProperty("clusterName");
+    userName = props.getProperty("userName");
+    password = props.getProperty("password");
+    containerPrefix = props.getProperty("table", "usertable") + "@";
+    String fieldcount = props.getProperty("fieldcount");
+    String fieldlength = props.getProperty("fieldlength");
+
+    LOGGER.info("notificationAddress=" + notificationAddress + " notificationPort=" + notificationPort +
+            " notificationMember=" + notificationMember);
+    LOGGER.info("clusterName=" + clusterName + " userName=" + userName + " password=" + password);
```

Comment:
generally speaking, one should not log passwords.

Reply:
I cut password log by 86f7c71a844c10ad69ed677f74f644bece58ba7b .